nmpolicy/internal/resolver/resolver.go

```diff
@@ -231,6 +235,12 @@ func (r resolver) wrapErrorWithCurrentExpression(err error) error {
 	if err == nil {
 		return nil
 	}
+
+	var pathError PathError
+	if errors.As(err, &pathError) {
+		r.currentNode = &pathError.errorNode
```

Comment:
You don't need to change `resolver` state also, the resolver is "by value" at the reciver let's just do the following it will be less confusing:

```golang
    errorNode := r.currentNode
    if errors.As(err, &pathError) {
		errorNode = &pathError.errorNode
    }
    if errorNode == nil || r.currentExpression == nil {
		return err
    }
    return expression.WrapError(err, *r.currentExpression, errorNode.Meta.Position)
```

Reply:
Since it is "by value` it can be changed and it will effect only the current method.
But I"m ok with both versions, so done.